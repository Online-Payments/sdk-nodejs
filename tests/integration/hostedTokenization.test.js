const sdkInstance = require('./__fixture__');
const config = require('./__fixture__/config');

describe('Hosted tokenization', () => {
  describe('Promises', () => {
    test('it can create a hosted tokenization', async () => {
      const sdkResponse = await sdkInstance.hostedTokenization.createHostedTokenization(config.merchantId, {}, {});
      expect(sdkResponse.isSuccess).toBe(true);
      expect(sdkResponse).toMatchSnapshot();
    });

    test('it can get a hosted tokenization', async () => {
      const sdkResponse = await sdkInstance.hostedTokenization.getHostedTokenization(
        config.merchantId,
        'HOSTED_TOKENIZATION_ID',
        {}
      );

      expect(sdkResponse.isSuccess).toBe(true);
      expect(sdkResponse).toMatchSnapshot();
    });
  });

  describe('Callback', () => {
    test('it can create a hosted tokenization', (done) => {
      sdkInstance.hostedTokenization.createHostedTokenization(config.merchantId, {}, {}, (error, sdkResponse) => {
        expect(error).toBeNull();
        expect(sdkResponse.isSuccess).toBe(true);
        expect(sdkResponse).toMatchSnapshot();
        done();
      });
    });

    test('it can get a hosted tokenization', (done) => {
      sdkInstance.hostedTokenization.getHostedTokenization(
        config.merchantId,
        'HOSTED_TOKENIZATION_ID',
        {},
        (error, sdkResponse) => {
          expect(error).toBeNull();
          expect(sdkResponse.isSuccess).toBe(true);
          expect(sdkResponse).toMatchSnapshot();
          done();
        }
      );
    });
  });
});
