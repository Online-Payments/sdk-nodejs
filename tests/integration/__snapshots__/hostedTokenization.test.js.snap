// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hosted tokenization Callback it can create a hosted tokenization 1`] = `
Object {
  "body": Object {
    "expiredCardTokens": Array [
      "0a7042d6-bf24-40b6-85c5-dec24b9dd1d1",
      "b7042dff-1fbd-46d2-9ecb-bddaa2c31165",
    ],
    "hostedTokenizationId": "string",
    "hostedTokenizationUrl": "https://payment.domain.com/hostedtokenization/tokenization/form/e18f459488504991858dcf624a94257f",
    "invalidTokens": Array [
      "0a7042d6-bf24-40b6-85c5-dec24b9dd1d1",
      "b7042dff-1fbd-46d2-9ecb-bddaa2c31165",
    ],
    "partialRedirectUrl": "string",
  },
  "isSuccess": true,
  "status": 200,
}
`;

exports[`Hosted tokenization Callback it can get a hosted tokenization 1`] = `
Object {
  "body": Object {
    "token": Object {
      "card": Object {
        "alias": "483247XXXXXX0249",
        "data": Object {
          "cardWithoutCvv": Object {
            "cardNumber": "string",
            "cardholderName": "string",
            "expiryDate": "0529",
          },
        },
      },
      "eWallet": Object {
        "alias": "customer-account@email.com",
        "customer": Object {
          "billingAddress": Object {
            "additionalInfo": "floor 9",
            "city": "Zaventem",
            "countryCode": "BE",
            "houseNumber": 3,
            "state": "string",
            "street": "Da Vinci street",
            "zip": 1930,
          },
          "companyInformation": Object {
            "name": "Customer Company Name",
          },
          "personalInformation": Object {
            "name": Object {
              "firstName": "John",
              "surname": "Doe",
            },
          },
        },
      },
      "externalTokenLinked": Object {
        "ComputedToken": "string",
        "GTSComputedToken": "string",
        "GeneratedToken": "string",
      },
      "id": "43763dd9-6b6b-4b05-914a-5953b14dcbba",
      "isTemporary": true,
      "paymentProductId": 840,
    },
    "tokenStatus": "UNCHANGED",
  },
  "isSuccess": true,
  "status": 200,
}
`;

exports[`Hosted tokenization Promises it can create a hosted tokenization 1`] = `
Object {
  "body": Object {
    "expiredCardTokens": Array [
      "0a7042d6-bf24-40b6-85c5-dec24b9dd1d1",
      "b7042dff-1fbd-46d2-9ecb-bddaa2c31165",
    ],
    "hostedTokenizationId": "string",
    "hostedTokenizationUrl": "https://payment.domain.com/hostedtokenization/tokenization/form/e18f459488504991858dcf624a94257f",
    "invalidTokens": Array [
      "0a7042d6-bf24-40b6-85c5-dec24b9dd1d1",
      "b7042dff-1fbd-46d2-9ecb-bddaa2c31165",
    ],
    "partialRedirectUrl": "string",
  },
  "isSuccess": true,
  "status": 200,
}
`;

exports[`Hosted tokenization Promises it can get a hosted tokenization 1`] = `
Object {
  "body": Object {
    "token": Object {
      "card": Object {
        "alias": "483247XXXXXX0249",
        "data": Object {
          "cardWithoutCvv": Object {
            "cardNumber": "string",
            "cardholderName": "string",
            "expiryDate": "0529",
          },
        },
      },
      "eWallet": Object {
        "alias": "customer-account@email.com",
        "customer": Object {
          "billingAddress": Object {
            "additionalInfo": "floor 9",
            "city": "Zaventem",
            "countryCode": "BE",
            "houseNumber": 3,
            "state": "string",
            "street": "Da Vinci street",
            "zip": 1930,
          },
          "companyInformation": Object {
            "name": "Customer Company Name",
          },
          "personalInformation": Object {
            "name": Object {
              "firstName": "John",
              "surname": "Doe",
            },
          },
        },
      },
      "externalTokenLinked": Object {
        "ComputedToken": "string",
        "GTSComputedToken": "string",
        "GeneratedToken": "string",
      },
      "id": "43763dd9-6b6b-4b05-914a-5953b14dcbba",
      "isTemporary": true,
      "paymentProductId": 840,
    },
    "tokenStatus": "UNCHANGED",
  },
  "isSuccess": true,
  "status": 200,
}
`;
